arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -173,49 +261,100 @@ ::filter( kwiver::vital::image_container_sptr input_image )
-        vxl::image_container::vital_to_vxl( color_commonality->get_image() ) );
+    auto color_commonality = convert_to_typed_vil_image_view< pix_t >(
+      color_commonality_filter->filter( input_image ) );
+
+    filtered_images.push_back( color_commonality );
   }
   if( enable_high_pass_box )
   {
-    auto high_pass_box = high_pass_box_filter->filter( input_image );
-    // 2 channels
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( high_pass_box->get_image() ) );
+    auto high_pass_box = convert_to_typed_vil_image_view< pix_t >(
+      high_pass_box_filter->filter( input_image ) );
+
+    // Legacy BurnOut models expect these channels to be incorrectly ordered
+    // TODO Remove this code when we no longer need to train models using
+    // legacy code
+    auto first_plane = vil_plane( high_pass_box, 0 );
+    auto second_plane = vil_plane( high_pass_box, 1 );
+    auto temp = vil_copy_deep( first_plane );
+    first_plane.deep_copy( second_plane );
+    second_plane.deep_copy( temp );
+
+    // 3 channels
+    filtered_images.push_back( high_pass_box );
   }
   if( enable_high_pass_bidir )
   {
-    auto high_pass_bidir = high_pass_bidir_filter->filter( input_image );
-    // 2 channels
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( high_pass_bidir->get_image() ) );
+    auto high_pass_bidir = convert_to_typed_vil_image_view< pix_t >(
+      high_pass_bidir_filter->filter( input_image ) );
+    // 3 channels
+    filtered_images.push_back( high_pass_bidir );
   }
+
+  vil_image_view< double > double_variance;
+  if( enable_average || enable_normalized_variance )
+  {
+    auto multiplane_variance = convert_to_typed_vil_image_view< double >( input_image );
+    if( multiplane_variance.nplanes() == 3 )
+    {
+      vil_convert_planes_to_grey( multiplane_variance, double_variance );
+    }
+    else
+    {
+      vil_math_mean_over_planes( multiplane_variance, double_variance );
+    }
+  }
+
   // TODO consider naming this variance since that option is used more
   if( enable_average )
   {
-    // 3 channels
-    auto averaged = average_frames_filter->filter( input_image );
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( averaged->get_image() ) );
+    auto variance = clamping_cast< pix_t >( double_variance );
+
+    // 1 channel
+    filtered_images.push_back( variance );
   }
   if( enable_aligned_edge )
   {
-    auto aligned_edge = aligned_edge_detection_filter->filter( input_image );
-    // 2 channels
-    filtered_images.push_back(
-        vxl::image_container::vital_to_vxl( aligned_edge->get_image() ) );
+    auto aligned_edge = convert_to_typed_vil_image_view< pix_t >(
+      aligned_edge_detection_filter->filter( input_image ) );
+
+    auto joint_response =
+      vil_plane( aligned_edge, aligned_edge.nplanes() - 1 );
+    // 1 channel
+    filtered_images.push_back( joint_response );
+  }
+  if( enable_normalized_variance )
```

Comment:
It seems weird that this is even an option.  It's not really variance if you are not normalizing by the number of frames.  That should be part of the variance calculation.  However, I understand that you are trying to match the old burnout tool first, so this doesn't need to be changed.

Reply:
I'd agree that the other variance option is quite odd. In practice, I believe it begins to hit the limit of the `char` range for many regions quite quickly.